htlcswitch/hop/payload.go

```diff
@@ -341,6 +365,33 @@ func ValidateParsedPayloadTypes(parsedTypes tlv.TypeMap,
 			Violation: IncludedViolation,
 			FinalHop:  isFinalHop,
 		}
+
+	// Blinded routes restrict the presence of TLVs more strictly than
+	// regular routes, check that intermediate and final hops only have
+	// the TLVs the spec allows them to have.
+	case hasEncryptedData:
+		allowedTLVs := map[tlv.Type]bool{
+			record.EncryptedDataOnionType: true,
+			record.BlindingPointOnionType: true,
+		}
+
+		if isFinalHop {
+			allowedTLVs[record.AmtOnionType] = true
+			allowedTLVs[record.LockTimeOnionType] = true
+			allowedTLVs[record.TotalAmtMsatBlindedType] = true
+		}
```

Comment:
Just thought about it because we also leave away the receiver TLVs in the encrypted data and it allows those TLV values even if we don't support receiving, but really up to you!

Reply:
Going to leave this as-is, but fully acknowledge it's pretty inconsistent :')

Reply:
if we are leaving this in, i think we might need to add 
`allowedTLVs[record.MPPOnionType] = true`

since we do allow payment addr to be set right? 

Ran into this when working on receiver side stuff.

Reply:
There's a new [total_amount](https://github.com/lightning/bolts/blob/60de4a09727c20dea330f9ee8313034de6e50594/04-onion-routing.md?plain=1#L316) field and [path_id](https://github.com/lightning/bolts/blob/60de4a09727c20dea330f9ee8313034de6e50594/04-onion-routing.md?plain=1#L547) for blinded MPP.

Reply:
gotcha